core/common/src/main/java/alluxio/network/TieredIdentityFactory.java

```diff
@@ -0,0 +1,123 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.network;
+
+import alluxio.Configuration;
+import alluxio.PropertyKey;
+import alluxio.PropertyKey.Template;
+import alluxio.util.ShellUtils;
+import alluxio.wire.TieredIdentity;
+import alluxio.wire.TieredIdentity.LocalityTier;
+
+import com.google.common.base.Preconditions;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import javax.annotation.Nullable;
+
+/**
+ * Class for getting tiered identity.
+ */
+public final class TieredIdentityFactory {
+  private static TieredIdentity sInstance = null;
+
+  /**
+   * @return the singleton tiered identity instance for this JVM
+   */
+  public static TieredIdentity getInstance() {
+    synchronized (sInstance) {
+      if (sInstance == null) {
+        sInstance = create();
+      }
+      return sInstance;
+    }
```

Comment:
Any point in making this more efficient when `mInstance != null`?

Reply:
might as well, updated